### PR TITLE
Native promises, server URL in dashboard, more config options, and tests

### PR DESCRIPTION
**Pull request by Conroman16**
Hey there DasKeyboard.  I've made significant updates to the Server Ping applet in the past couple weeks.  I made a PR to the [parent repo](https://github.com/chbakouras/daskeyboard-applet--server-ping) and it was approved/merged, [however there seems to be no movement toward merging the most-recent changes into the marketplace repo](https://github.com/daskeyboard/q-marketplace/issues/14).  Therefore I'd like to propose a merge from the stable master branch of the [parent repo](https://github.com/chbakouras/daskeyboard-applet--server-ping) into the marketplace repo, as the parent repo contains the most-recent stable version of the Server Ping applet.

### CODE REVIEW

**Comment by Conroman16** [comment]
I should also add, that @chbakouras has requested a merge [here](https://github.com/daskeyboard/q-marketplace/issues/14) in Issue #14 on the q-marketplace repo but it hasn't gotten any attention which is why I'm submitting this PR here in the first place

**Comment by Conroman16** [comment]
@Mubramaj Since you were the one who merged my last update on the ICMP applet, I wanted to call you out here and ask, might you guys be able to take a look at this one?  I know you guys are busy over there making daskeyboard awesome, but it would be fantastic to see this update be released into the marketplace sooner than later.

**Comment by Conroman16** [comment]
Lil' Dicky Is Anybody Out There GIF

**Comment by Mubramaj** [comment]
Hi @Conroman16 I am sorry for the delay. Looking into it right now

File: package.json [hunk 1]
```diff
@@ -33,32 +39,58 @@
       "height": 1
     },
     "applet": {
-      "defaults": {}
+      "defaults": {
+        "requestInterval": 60,
+        "upColor": "#00ff00",
+        "downColor": "#ff0000",
+        "desiredStatusCode": 200
+      }
```
Comment (Mubramaj) [inline package.json]: This is not used. To set default values use `value` attribute  of each question.  Example: 

```
      {
        "key": "requestInterval",
        "label": "Server ping frequency (seconds)",
        "help": "How often (in seconds) to ping the server URL",
        "required": false,
        "order": 20,
        "controlType": "textbox",
        "placeholder": "60"
        "value": "60"  <--------------------------
      },
```

File: package.json [hunk 2]
```diff
@@ -1,27 +1,33 @@
 {
-  "name": "q-applet-server-ping",
+  "name": "daskeyboard-applet--server-ping",
   "displayName": "Server Ping",
-  "version": "1.0.0",
```
Comment (Mubramaj) [inline package.json:4]: Do you want users to rewalk through the configuration when upgrading or no ? 

If you want the user to reconfigure there applets when upgrading you need to increment the major version. E.g 2.0.0 
If not you need to make sure that the new index.js will work with the old `questions` keys

**Review by Mubramaj — CHANGES_REQUESTED** [review]
It looks good. Thanks for adding the tests. Just a couple changes then I will merge it

**Comment by Conroman16** [comment]
Hey apologies, hadn't seen that you got back to me there.  Anyways, I made the changes and put in [this PR](https://github.com/chbakouras/daskeyboard-applet--server-ping/pull/10) for @chbakouras to merge and then we should be good

File: package.json (hunk 2 above)
Comment (Conroman16) [reply package.json:4]: I reviewed it and I don't think they do.  I kept all the original configuration elements intact and added one, but the applet is self-sustaining in terms of default configuration so it's fine if they don't configure the new `desiredStatusCode` param

File: package.json (hunk 1 above)
Comment (Conroman16) [reply package.json]: I just went ahead and pulled the whole defaults section and did not add any `value` params to the config blocks as I wrote the applet's configuration to be self-sustaining with the same defaults as in that block.  I just for some reason thought I needed that section.  At this point the applet could run with no other configuration than a url, and that configuration item didn't change for this revision

File: package.json (hunk 2 above)
Comment (Conroman16) [reply package.json:4]: The status code used to be hard-coded to `200`.  Now it defaults to `200` but can also be changed if the user so desires

**Comment by chbakouras** [comment]
@Conroman16 PR is merged, so we are good to go! @Mubramaj could you please review the requested changes?

**Pull request merged**